### PR TITLE
dosfstools: fix build failure in darwin

### PR DESCRIPTION
**Pull request by otavio**
Log of error: https://hydra.nixos.org/build/175255572/nixlog/1
Refs: #172160
Signed-off-by: Otavio Salvador <otavio@ossystems.com.br>

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/filesystems/dosfstools/default.nix [hunk 1]
```diff
@@ -11,6 +11,13 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-2gxB0lQixiHOHw8uTetHekaM57fvUd9zOzSxWnvUz/c=";
   };
 
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/dosfstools/dosfstools/commit/77ffb87e8272760b3bb2dec8f722103b0effb801.patch";
```
Comment (thiagokokada) [inline pkgs/tools/filesystems/dosfstools/default.nix:16]: Can you add a comment explaining where you got this patch (from a PR, from `master`, etc)?

Comment (otavio) [reply pkgs/tools/filesystems/dosfstools/default.nix:17]: done.

**Comment by thiagokokada** [comment]
Please rebase.

**Review by Mindavi — APPROVED** [review]
Diff / patch LGTM

**Comment by thiagokokada** [comment]
Result of `nixpkgs-review pr 172212` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dosfstools</li>
  </ul>
</details>

**Review by thiagokokada — APPROVED** [review]

**Comment by thiagokokada** [comment]
Tested and seems to be working fine.

**Comment by thiagokokada** [comment]
@ofborg eval

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 172212` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>dosfstools</li>
  </ul>
</details>

**Pull request merged**